### DISCUSSION TITLE
autodiscoverExclusions config property

### DISCUSSION THREAD

**Discussion by markchoward7** [renovatebot/renovate#25812] [Suggest an Idea]
### Tell us more.

As mentioned in #21347 and to support #25786 

Create a new config option called `autodiscoverExclusions` which works in the same way as `autodiscoverFilter`, but subtracts matching repositories from the result.

**Comment by rarkins** [comment]
@markchoward7 can you take a look at https://github.com/renovatebot/renovate/pull/24695 and let us know if it would satisfy your use case?

**Reply by markchoward7** [reply]
Possibly, I want to make sure I understand the proposal.

If the filters were like so: `["mygroup/*", "!not-this-one"]` and my repos were: `mygroup/this-one`, `mygroup/not-this-one`, and `othergroup/this-one`. Would that just return `mygroup/this-one`? If so, then it should work.

**Reply by rarkins** [reply]
Yes

**Reply by markchoward7** [reply]
Awesome, then I've went ahead and marked this as the answer as I believe it should work.

**Reply by rarkins** [reply]
Did you test your own PR locally? If so then you should be able to test like locally too

**Reply by markchoward7** [reply]
I hadn't set up repos to test with, I can probably find some time to do that though.

**Reply by stefan-g** [reply]
@rarkins does the negate filter works like that ["mygroup/*", "!not-this-one"]?  
According to https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter only a single negate filter can be used and should not be mixed with other filters.

**Reply by rarkins** [reply]
We'll improve this in the upcoming v38 so that it supports multiple negating filters, and both minimatch and regex

**Reply by alsyia** [reply]
Has this been implemented since? The doc still says only one negative filter is supported and I couldn't find another parameter :/

**Reply by rarkins** [reply]
Not yet, but I have created a PR for it to be included in the next major release: https://github.com/renovatebot/renovate/pull/34707
